File: Examples/Features/Workbook/ImportCsvFileToWorkbook.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Workbook
{
    public class ImportCsvFileToWorkbook : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            Stream stream = this.GetResourceStream("xlsx\\Information.csv");

            //Open csv file stream.
            workbook.Open(stream, OpenFileFormat.Csv);
        }

        public override string TemplateName
        {
            get
            {
                return "Information.csv";
            }
        }

        public override bool CanDownload
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return true;
            }
        }
        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\Information.csv"};
            }
        }
    }

}

File: Examples.VB/Features/Workbook/ImportCsvFileToWorkbook.vb
Imports System.IO
Namespace Features.Workbook
    Public Class ImportCsvFileToWorkbook
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim stream As Stream = GetTemplateStream()
            'Open csv file stream.
            workbook.Open(stream, OpenFileFormat.Csv)
        End Sub
        Public Overrides ReadOnly Property TemplateName As String
            Get
                Return "Information.csv"
            End Get
        End Property
        Public Overrides ReadOnly Property CanDownload As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
